### ISSUE TITLE
[Crash] RaycastManager in WebUI events

### ISSUE THREAD

**Issue by FlashHit** [VeniceUnleashed/VeniceUnleashed#793]
Repro:
[ReproRaycastCrash.zip](https://github.com/EmulatorNexus/VeniceUnleashed/files/8770834/ReproRaycastCrash.zip)
- join server
- press red button
-> Crash
```
Crash ID: 29c35bcf-f386-4586-a834-cc6c030475f5
Build Number: 18808
Release Branch: dev
```
```
---------------------------
vu
---------------------------
[string "client.lua"]:49: <unknown>

---------------------------
fastload
---------------------------
[string "__init__.lua"]:32: <unknown>

---------------------------
reproraycastcrash
---------------------------
[string "__init__.lua"]:7: _DoRaycast
```

This gets called in a WebUI event and I guess this is in a wrong UpdatePass(?). Shouldn't crash tho.
Can we prevent it from crashing and instead throw an error or something?

**Comment by OrfeasZ** [comment]
Tried a couple of things, but I don't think there's anything I can do about that currently since I dont track the current thread / update pass. Will keep open as low-prio.

**Comment by FlashHit** [comment]
@OrfeasZ the update affected MapEditor which did the client raycasts on `Engine:Update` instead of in `UpdateManager:Update` in updatepass `UpdatePass_PreSim`. It resulted in a crash. I fixed the crash by doing the Raycast only in `UpdatePass_PreSim`, so now it works again.

Anyways here the crash report:
```
Ausnahme ausgelöst bei 0x0041BE5C in vu.exe: 0xC0000005: Zugriffsverletzung beim Schreiben an Position 0x0000000C.

Crash ID: 9c8adef8-5d24-4c6f-b1a0-f6763bc8675c
Build Number: 18848
Release Branch: dev

---------------------------
vu
---------------------------
[string "client.lua"]:188: <unknown>

---------------------------
fastload
---------------------------
:0: <unknown>

---------------------------
mapeditor
---------------------------
[string "Editor.lua"]:76: Raycast
```

Repro:
https://github.com/BF3RM/MapEditor/tree/f725e92dac29d0a0c7f4d0d73fdd9746fe0db546
- Map shouldn't matter but I tried with `XP3_Shield ConquestLarge0 1`
- Join with client -> straight crash

**Comment by FlashHit** [comment]
This was fixed in build `19968` / `v1.12.0`
